### PR TITLE
[CTF Rebuild] Fire arrows now light team kegs as well

### PR DESCRIPTION
**Pull request by bunniewormy**
## Status

(pick one)

- **READY**: this PR is (to the best of your knowledge) ready to be incorporated into the game.

## Description

[changed] fire arrows can now light team kegs

### CODE REVIEW

**Comment by MonkeyFeats** [comment]
Not sure about this, I see what you are going for, but I think it will piss people off more than anything.

**Comment by bunniewormy** [comment]
> 
> 
> Not sure about this, I see what you are going for, but I think it will piss people off more than anything.

How will it "piss people off"? It's an offensive buff to archers, to make them able to use kegs if they spend additional money on a fire arrow.

**Review by Vam-Jam — DISMISSED** [review]
code looks good

**Comment by CatWiz** [comment]
This lets you set kegs carried by teammates on fire. Need to check for carrier blob's (or keg's) team.

**Comment by CatWiz** [comment]
Also fire arrow should break when hitting keg instead of bouncing off it.

**Comment by bunniewormy** [comment]
Fixed - fire arrow now breaks when hitting keg and doesn't set kegs held by teammates on fire

File: Entities/Items/Projectiles/Arrow.as [hunk 1]
```diff
@@ -333,8 +333,9 @@ bool doesCollideWithBlob(CBlob@ this, CBlob@ blob)
 		return true;
 	}
 
-	//definitely collide with non-team blobs
-	bool check = this.getTeamNum() != blob.getTeamNum() || blob.getName() == "bridge";
+	bool check = this.getTeamNum() != blob.getTeamNum() || // collide with all enemy kegs
+	blob.getName() == "bridge" ||
```
Comment (Mazey) [inline Entities/Items/Projectiles/Arrow.as:337]: Spacing; align 337/338 with 336 `this`.

File: Entities/Items/Projectiles/Arrow.as [hunk 2]
```diff
@@ -484,6 +485,7 @@ f32 ArrowHitBlob(CBlob@ this, Vec2f worldPoint, Vec2f velocity, f32 damage, CBlo
 
 		// check if shielded
 		const bool hitShield = (hitBlob.hasTag("shielded") && blockAttack(hitBlob, velocity, 0.0f));
+		const bool hitKeg = hitBlob.getName() == "keg";
```
Comment (Mazey) [inline Entities/Items/Projectiles/Arrow.as:488]: Put the condition in brackets like with `hitShield` for consistency, KAG does this (almost) always when setting a variable to a condition.

**Review by Mazey — APPROVED** [review]

File: Entities/Items/Projectiles/Arrow.as (hunk 1 above)
Comment (bunniewormy) [reply Entities/Items/Projectiles/Arrow.as]: resolved

File: Entities/Items/Projectiles/Arrow.as (hunk 2 above)
Comment (bunniewormy) [reply Entities/Items/Projectiles/Arrow.as]: resolved

**Review by AsuMagic — DISMISSED** [review]

**Review by AsuMagic — APPROVED** [review]
No functional change since previous approval

**Review by AsuMagic — APPROVED** [review]

**Pull request merged**